test/ruby_lsp_rails/document_symbol_test.rb

```diff
@@ -155,6 +155,19 @@ class NestedTest < ActiveSupport::TestCase
         assert_equal("back to the same level", response[0].children[2].name)
       end
 
+      test "correctly handles attr_readonly" do
+        response = generate_document_symbols_for_source(<<~RUBY)
+          class FooModel < ApplicationRecord
+            attr_readonly :foo
+          end
+        RUBY
+
+        assert_equal(1, response.size)
+        assert_equal("FooModel", response[0].name)
+        assert_equal(1, response[0].children.size)
+        assert_equal("attr_readonly :foo", response[0].children[0].name)
+      end
+
```

Comment:
I wasn't sure if you wanted this test. I see that has_one, has_many, has_and_belongs_to_many are all untested and just fall under the `belongs_to` test basically. So maybe this test isn't necessary.

Reply:
Yeah, it's not essential, but I'm ok with keeping it for now.